torchtitan/checkpoint.py

```diff
@@ -160,19 +211,125 @@ def save(self, curr_step: int, force: bool = False) -> None:
                     k: v.to(self.export_dtype) for k, v in self.states.items()
                 }
             logger.info(
-                f"Saving a model weights only checkpoint in {self.export_dtype} at step {curr_step}"
+                f"Saving a model weights only checkpoint in {self.export_dtype} "
+                f"at last step, step {curr_step}."
             )
-
         else:
-            logger.info(f"Saving a full checkpoint at step {curr_step}")
+            logger.info(f"Saving a full checkpoint at last step, step {curr_step}.")
 
-        begin = time.monotonic()
         dcp.save(self.states, checkpoint_id=self._create_checkpoint_id(curr_step))
         self.reset()
+
+    def _should_save(self, curr_step: int, force: bool = False) -> bool:
+        if not self.enable_checkpoint:
+            return False
+
+        if not force:
+            if self.interval_type == IntervalType.STEPS and not (
+                curr_step % self.interval == 0
+            ):
+                return False
+            if self.interval_type == IntervalType.SECONDS:
+                time_sync_result = (time.monotonic() - self.begin_time) >= self.interval
+                self.time_sync_result = torch.tensor(int(time_sync_result))
+                if self.time_sync_work is None:
+                    self.time_sync_work = dist.all_reduce(
+                        self.time_sync_result, group=self.pg, async_op=True
+                    )
+                    return False
+                elif curr_step % 5 == 4:
```

Comment:
I guess 5 seconds is a good default for synching, but should this be exposed as a class or var ala _interval_sync_step = 5, and then it becomes 
curr_step % _interval_sync_step = _interval_sync_step -1?

Reply:
This time interval was added before, we can change it in a separate PR.